dbt/include/athena/macros/dbt_utils/get_tables_by_pattern_sql.sql

```diff
@@ -0,0 +1,16 @@
+{% macro athena__get_tables_by_pattern_sql(schema_pattern, table_pattern, exclude='', database=target.database) %}
+
+    {% set sql %}
+        select
+            table_schema as "table_schema",
+            table_name as "table_name",
+            {{ dbt_utils.get_table_types_sql() }}
```

Comment:
Does the inclusion of a `dbt_utils` reference in the main adapter cause an issue if this adapter is installed without `dbt_utils` also installed?

Reply:
I guess so, this requires the package and force the run of dbt deps no? can we have `get_table_types_sql` in the adapter itself? IMHO we shouldn't force usage of specific packages.

Reply:
I tested this and my project compiled without issue when this macro was present.

Reply:
...but to your point @nicor88, maybe the best course of action is to have the adapter support [these macros](https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros) and keep other package compatibility shims in a separate project (e.g. [athena-utils](https://github.com/lalalilo/athena-utils))?

Reply:
@owenprough-sift I didn't test it, but if you said that it works, then we are fine.

regarding https://github.com/dbt-athena/dbt-athena/pull/96#discussion_r1054723192 
I think that is a good idea to don't keep essentials macro in the adapter. Something to consider in the future.

Reply:
Shall I therefore close this PR and move #96 to the `athena-utils` repo?

Reply:
Honestly, if it works I will leave like that for now.

Reply:
I actually don't know if this implementation _does_ work. It doesn't crash, but I couldn't get `dbt_utils.get_relations_by_pattern()` to recognize this macro.
I tried updating my project's dispatch config to:
```
dispatch:
  - macro_namespace: dbt_utils
     search_order: [ "dbt-athena", "dbt_athena", "dbt_utils"]
```

...but a model containing `{% set all_tables = dbt_utils.get_relations_by_pattern('%', '%') %}` failed with `mismatched input 'ilike'` (the error that should be avoided by this PR's changes). So I am uncertain how to get dbt to recognize this macro's override in the adapter.

Weirdly, it works fine if I add the macro to my project and then update dispatch config to list my project first in the search order.

Reply:
I think you may need to register dbt_athena as a package also ?
If so, I would suggest moving things to an other package, maybe `athena-utils` but if we want to make things clean, maybe `dbt-athena-packages` or something like that ?

Reply:
I'm going to move this to `athena-utils` so that the dbt-athena adapter code doesn't need to know about dbt-utils.

Reply:
👍🏻